### DISCUSSION TITLE
Global export not available in main.rs

### DISCUSSION THREAD

**Discussion by FoundationKen** [slint-ui/slint#3511] [Q&A] [answered]
I am following the docs here to create a global callback function :

    https://slint.dev/releases/1.2.0/docs/slint/src/recipes/recipes#global-callbacks

I have a slint file with the following declaration:

```Rust
export global Utils {
    pure callback color-to-hex(color) -> string;
}
```

My `main` looks like this:

```Rust
slint::include_modules!();

fn main() {
    let app = App::new().unwrap();

    app.global::<Utils>().on_color_to_hex(|color| {
        "#ff00ff"
    });

    app.run().unwrap();
}
```

I was able to reference the `Utils.color-to-hex()` function in one of my other slint files, and it does not give an error in VSCode while editing, but I get the following compiler error on the Rust side:

```
error[E0412]: cannot find type `Utils` in this scope
  --> src/main.rs:20:18
   |
20 |     app.global::<Utils>().on_color_to_hex(|color| {
   |                  ^^^^^ not found in this scope
```

I was expecting that the `slint::include_modules!();` would bring the globals into scope the way it does for `App`, which is in my local `slint!()` macro, but that does not seem to be happening.

Is there some step I am missing to make `Utils` available in the Rust code?

**Comment by ogoffart** [comment, marked as answer]
you should do re-import and `export { Utils }` from the main slint file (or slint! macro)

**Reply by FoundationKen** [reply]
Thanks!  That is the fix.

**Reply by kyrilldev** [reply]
Thank you so much!
